rosidl_generator_c/resource/msg__struct.h.em

```diff
@@ -129,10 +141,29 @@ enum
 @#>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
 
 @#<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
-// Struct defined in @(interface_path_to_string(interface_path)) in the package @(package_name).
+/// Struct defined in @(interface_path_to_string(interface_path)) in the package @(package_name).
+@{comments = message.structure.get_comment_lines()}@
+@[if comments]@
+/**
+@[  for line in comments]@
+@[    if line]@
+  * @(line)
+@[    else]@
+  *
+@[    end if]@
+@[  end for]@
+ */
+@[end if]@
 typedef struct @(idl_structure_type_to_c_typename(message.structure.namespaced_type))
 {
 @[for member in message.structure.members]@
+@[  for line in member.get_comment_lines()]@
+@[    if line]@
+  /// @(line)
+@[    else]@
+  ///
```

Comment:
IIUC, this is inserting an empty comment. Why do this versus inserting nothing?

If this is to satisfy linters, we could alternatively be explicit about there being no documentation.
E.g. `/// Not documented` or some identifier that in theory someone could use to detect something is not documented.

Reply:
Same for the struct comment above.

Reply:
The idea is to respect the new lines in the original comment.
If I delete the `if/else` and just use `/// @(line)`, empty lines will generate `/// ` (note the space at the end), and linters will reject the generated header file.
For empty lines I want to leave an empty line in the comments, which equivalent is `///`.

We could also ommit new lines, but I think it's better to maintain them.

Reply:
Thanks for explaining. That makes sense to me :+1: